dsl_parser/tests/test_functions.py

```diff
@@ -1251,109 +1299,99 @@ def test_func_integrations_within_list(self):
+                "The recursion limit \\([0-9]+\\) has been reached while "
+                "evaluating the deployment\\..+get_property\\[0\\].+"):
+            prepare_deployment_plan(self.parse(yaml))
 
+    def test_circular_within_dict(self):
         yaml = """
-inputs:
-    dummy_input:
-        default: some_node
 node_types:
     some_type:
         properties:
             prop: {}
-            dummy_prop: {}
 node_templates:
     some_node:
         type: some_type
         properties:
-            dummy_prop: dummy_value
             prop:
-                get_attribute:
-                    - get_input: dummy_input
-                    - concat:
-                        - get_attribute: [ some_node, dummy_prop ]
-                    - get_input:
-                        concat:
-                            - get_attribute: [ some_node, dummy_prop ]
-                            - get_input: dummy_input
-    """
+                key:
+                    get_property:
+                        - get_property:
+                                - some_node
+                                - prop
+                        - shouldn't matter
+            """
         with self.assertRaisesRegexp(
-                exceptions.FunctionValidationError,
-                'Runtime function .+ cannot be nested within a non-runtime '
-                'function \\(found in .+\\)'):
-            prepare_deployment_plan(self.parse_1_1(yaml))
+                exceptions.EvaluationRecursionLimitReached,
+                "The recursion limit \\([0-9]+\\) has been reached while "
+                "evaluating the deployment\\..+get_property\\[0\\].+"):
+            prepare_deployment_plan(self.parse(yaml))
 
-    def test_get_property_doesnt_accept_runtime_func_as_args(self):
+    def test_circular_within_list(self):
         yaml = """
 node_types:
     some_type:
         properties:
             prop: {}
-            dummy_prop: {}
 node_templates:
     some_node:
         type: some_type
         properties:
-            dummy_prop: dummy_value
             prop:
-                get_property:
-                    - get_attribute:
+                - get_property:
+                    - get_property:
                             - some_node
-                            - dummy_prop
+                            - prop
                     - shouldn't matter
-    """
+            """
         with self.assertRaisesRegexp(
-                exceptions.FunctionValidationError,
-                'Runtime function .+ cannot be nested within a non-runtime '
-                'function \\(found in .+\\)'):
+                exceptions.EvaluationRecursionLimitReached,
+                "The recursion limit \\([0-9]+\\) has been reached while "
+                "evaluating the deployment\\..+get_property\\[0\\].+"):
             prepare_deployment_plan(self.parse(yaml))
 
+    def test_get_input_doesnt_accept_runtime_func_as_args(self):
```

Comment:
well I am still for multi test functions of quicker understanding of what's wrong when it will fail